pkg/apis/keycloak/v1alpha1/keycloak_types.go

```diff
@@ -79,6 +79,9 @@ type KeycloakSpec struct {
 	// Name of the StorageClass for Postgresql Persistent Volume Claim
 	// +optional
 	StorageClassName *string `json:"storageClassName,omitempty"`
+	//POD affinity rules for KeyCloak deployment
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
```

Comment:
I think we should probably hide the complexity under some nice option. Perhaps something like this:
```
apiVersion: keycloak.org/v1alpha1
kind: Keycloak
metadata:
  name: example-keycloak
  labels:
    app: sso
spec:
  instances: 1
  externalAccess:
    enabled: True
  nodeAntiAffinity:
    enabled: True
```

Thinking about this more, I think both `nodeAntiAffinity` and `podDisruptionBudget` should be always set to true (WDYT @wei-lee). It's hard to think about the scenarios where you don't want these.

Reply:
@slaskawi would this then reduce the flexibility for the user to specify his own settings/config/preferences for these? (in case he needs them).  Surely our default config cannot satisfy everyone's needs.

Reply:
I don't think it matters in this case. Multi AZ deployment are all similar.

But if you insist, we can make a compromise here. Let's introduce a `nodeAntiAffinity` setting as well as raw Affinity settings in the experimental section. Would that work for you?

Reply:
@slaskawi are we always mandating Multi AZ deployment? (sorry for my ignorance).
Curious to understand why we want them in the experimental section ( i assumed experimental section can change/be removed in the future). Wouldn't these affinity settings be more of a standard requirement if desired?

Reply:
After discussing with @slaskawi - Decided to hide the low level details from the user. The deployment CR would have ```  multiAvailablityZones:
    enabled: True``` that can enabled for multi Active zone deployments. This would create a keycloak deployment with the podAntiAffinity settings.

Reply:
Didn't we agree to put raw Affinity object in the experimental section?

Reply:
@slaskawi yes, i missed that. I will update the PR now.

Reply:
@slaskawi  : in a scenario where we have `multiAvailablityZones: enabled: True` and the user had defined the affinity object in the Experimental section, which one should be used (takes precedence?), I think it should be the explicit settings done by the user. Is that correct?

Reply:
Agreed with @slaskawi that the explicit settings done for affinity in the experimental section would take precedence even in the case of multiAvalabiltyZones being enabled.

Reply:
Works for me! Thanks!